driver-kafka/src/main/java/io/openmessaging/benchmark/driver/kafka/KafkaBenchmarkConsumer.java

```diff
@@ -56,7 +57,7 @@ public KafkaBenchmarkConsumer(KafkaConsumer<String, byte[]> consumer,
                                   long pollTimeoutMs) {
         this.consumer = consumer;
         this.executor = Executors.newSingleThreadExecutor();
-
+        this.autoCommit= Boolean.valueOf((String)consumerConfig.getOrDefault(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG,"false"));
```

Comment:
`autoCommit` isn't used, we need to replace `if (!offsetMap.isEmpty() && timeSinceOffsetCommit > TimeUnit.SECONDS.toNanos(5)) {` with `if (!autoCommit&&!offsetMap.isEmpty()) {`

Reply:
@rystsov - no, this will re-introduce a bug I just fixed which is that if `auto.commit` is not used we commit offsets for every single group of records consumed, which can be several times per millisecond. This results in a huge backlog of offset commits which cause a variety of problems.

We should have two modes: an `auto.commit` mode which just does nothing explicit (the `!autoCommit` part), like the Kafka driver, and then the "explicit mode" as we have now with a throttle of only making commits every N seconds, currently 5. Perhaps this part could be configurable.

Reply:
@travisdowns I'm not sure it's a bug. "commit offsets for every single group of records consumed" is a valid use case and OMB use this workload for other systems. If we hack the workload just for Redpanda it may seem as we're trying to rig the benchmarks. We should use the standard way by default and use configuration for special cases like this

Reply:
@rystsov - point taken about "fairness", but the test eventually fails in this scenario, because the offset commit backlog gets so large on the client side that it times out when some offset cannot be committed in 300,000 ms (5 minutes), because they have never reached the head of the queue.

Ultimately, the test numbers do not reflect the scenario anyway: the test "finishes" and then the measurements are collected, but then the workers spend several minutes committing a giant backlog of (possibly millions) of offsets, which isn't measured.

It's just broken. I don't think you'd ever design an application like this: if the queue has offset 10 already in it and you go to commit offset 20, there is no point in committing offset 10 the queue should "coalesce" and only commit 20. This can be implemented on the client side using a callback. The client library could also do such coalescing but currently it does not.

The other option is to use `commit` (not async). This prevents a backlog from forming and would be a valid test if you wanted to be sure that all offsets were committed for every batch of work.

I believe the Kafka driver has already effectively fixed this because they support `auto.commit` and bypass the explicit `commitAsync` in that case. Since auto.commit defaults to true I think the explicit commit mostly won't happen.